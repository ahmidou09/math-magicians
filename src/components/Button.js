import PropTypes from 'prop-types';

const getStyleName = (btn) => {
  const className = {
    '=': 'opt',
    'ร': 'opt',
    '-': 'opt',
    '+': 'opt',
    'รท': 'opt',
    0: 'zero',
  };
  return className[btn];
};

const Button = ({ value }) => (
  <button type="button" className={`${getStyleName(value)} button`}>
    {value}
  </button>
);
Button.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default Button;
