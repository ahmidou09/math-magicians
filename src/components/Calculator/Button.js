import React from 'react';
import PropTypes from 'prop-types';

const getStyleName = (btn) => {
  const className = {
    '=': 'opt',
    'ร': 'opt',
    '-': 'opt',
    '+': 'opt',
    'รท': 'opt',
    0: 'zero',
  };
  return className[btn];
};

const Button = ({ value, onClick }) => (
  <button type="button" className={`${getStyleName(value)} button`} onClick={onClick}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
