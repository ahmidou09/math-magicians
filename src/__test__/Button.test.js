/* eslint-disable linebreak-style */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../components/Calculator/Button';

test('renders Button component with value', () => {
  const { container } = render(<Button value="7" onClick={() => {}} />);
  expect(container).toMatchSnapshot();
});

test('fires onClick event when Button is clicked', () => {
  const mockOnClick = jest.fn();
  const { getByText } = render(<Button value="+" onClick={mockOnClick} />);
  fireEvent.click(getByText('+'));
  expect(mockOnClick).toHaveBeenCalled();
});
