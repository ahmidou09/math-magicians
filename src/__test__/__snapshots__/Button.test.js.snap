// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component with value 1`] = `
<div>
  <button
    class="undefined button"
    type="button"
  >
    7
  </button>
</div>
`;
